src/renderer/electron-types.ts

```diff
@@ -20,41 +25,82 @@ export class ElectronTypes {
 
   constructor(
     private readonly monaco: typeof MonacoType,
-    private readonly cacheDir: string,
+    private readonly electronCacheDir: string,
+    private readonly nodeCacheDir: string,
   ) {}
 
   public async setVersion(ver?: RunnableVersion): Promise<void> {
     this.clear();
 
     if (!ver) return;
+
+    await this.setElectronTypes(ver);
+    await this.setNodeTypes(ver.version);
+  }
+
+  private async setElectronTypes(ver: RunnableVersion): Promise<void> {
     const { localPath: dir, source, version } = ver;
 
+    // If it's a local development version, pull Electron types from out directory.
     if (dir) {
       const file = path.join(dir, 'gen/electron/tsc/typings', ELECTRON_DTS);
-      this.setTypesFromFile(file, ver);
+      this.setTypesFromFile(file, version);
       try {
-        this.watcher = watch(file, () => this.setTypesFromFile(file, ver));
+        this.watcher = watch(file, () => this.setTypesFromFile(file, version));
       } catch (err) {
         console.debug(`Unable to watch "${file}" for changes: ${err}`);
       }
     }
 
+    // If it's a published version, pull from cached file.
     if (source === VersionSource.remote) {
-      const file = this.getCacheFile(ver);
-      await ElectronTypes.ensureVersionIsCachedAt(version, file);
-      this.setTypesFromFile(file, ver);
+      const file = this.getCacheFile(version);
+      await this.ensureElectronVersionIsCachedAt(version, file);
+      this.setTypesFromFile(file, version);
     }
   }
 
+  private async setNodeTypes(version: string): Promise<void> {
+    // Get the Node.js version corresponding to the current Electron version.
+    const v = releases.find((release: Version) => {
+      return normalizeVersion(release.version) === version;
+    })?.node;
+
+    if (!v) return;
+
+    const dir = this.getCacheDir(v);
+    const ver = await this.cacheAndReturnNodeTypesVersion(v, dir);
+    await this.setTypesFromDir(dir, ver);
+  }
+
   public uncache(ver: RunnableVersion) {
     if (ver.source === VersionSource.remote)
-      fs.removeSync(this.getCacheFile(ver));
+      fs.removeSync(this.getCacheFile(ver.version));
   }
 
-  private setTypesFromFile(file: string, ver: RunnableVersion) {
+  private async setTypesFromDir(dir: string, version: string) {
     this.dispose();
+
     try {
-      console.log(`Updating Monaco with "${ELECTRON_DTS}@${ver.version}"`);
+      const files = await readdir(dir);
+      for (const file of files) {
+        if (file.endsWith('.d.ts')) {
+          const { base: filename } = path.parse(file);
+          console.log(`Updating Monaco with "${filename}@${version}"`);
+          this.disposable = this.monaco.languages.typescript.javascriptDefaults.addExtraLib(
```

Comment:
Should `disposable` be changed to `disposables`, an array of `Disposable`? It looks like now there can be multiple, and this might be leaking disposables added in a previous iteration of this for loop. So probably need to make it an array and change `dispose()` loop through it disposing of them.

Reply:
done!